hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java

```diff
@@ -109,15 +111,31 @@ public static void deleteMetadataTable(String basePath, HoodieEngineContext cont
             : hoodieWriteStat.getTotalWriteBytes();
         newFiles.put(filename, totalWriteBytes);
       });
+
       // New files added to a partition
-      HoodieRecord record = HoodieMetadataPayload.createPartitionFilesRecord(
-          partition, Option.of(newFiles), Option.empty());
-      records.add(record);
+      if (!newFiles.isEmpty()) {
+        HoodieRecord record = HoodieMetadataPayload.createPartitionFilesRecord(
+                partition, Option.of(newFiles), Option.empty());
+        records.add(record);
+      }
     });
 
-    // New partitions created
-    HoodieRecord record = HoodieMetadataPayload.createPartitionListRecord(new ArrayList<>(allPartitions));
-    records.add(record);
+    // Add delete partition's record
+    if (commitMetadata instanceof HoodieReplaceCommitMetadata
+            && WriteOperationType.DELETE_PARTITION.equals(commitMetadata.getOperationType())) {
```

Comment:
would it be better to extract these logic to a common util method to make it explicit about this check? say `isDeletePartition()`. Recall somewhere else also checking this

Reply:
can this refactoring be addressed somehow?